providers/shared/inputsources/shared/setting.ftl

```diff
@@ -0,0 +1,98 @@
+[#ftl]
+
+[#---------------------------------------------------------------
+-- Internal support functions for composite setting processing --
+-----------------------------------------------------------------]
+
+[#function internalReformatSettings objects fileRegex=".*"]
+    [#local result = {}]
+    [#list objects!{} as key,value]
+        [#local result =
+            mergeObjects(
+                result,
+                internalReformatFiles(key, value!{}, fileRegex)
+            )]
+    [/#list]
+    [#return result]
+[/#function]
+
+[#function internalReformatFiles key settingsFiles fileRegex]
```

Comment:
A comment here on the purpose of the reformat would be helpful. A lot of reading is required to understand the desired final format.

Reply:
I moved these functions from composite to the shared provider so they would be available to both providers. Writing these functions wasn't part of this change